Lib/enum.py

```diff
@@ -731,14 +731,18 @@ def __contains__(cls, value):
         """
         if isinstance(value, cls):
             return True
-        try:
-            cls(value)
-            return True
-        except ValueError:
-            return (
-                    value in cls._unhashable_values_    # both structures are lists
-                    or value in cls._hashable_values_
-                    )
+        if issubclass(cls, Flag):
+            try:
+                result = cls._missing_(value)
+                if isinstance(result, cls):
+                    return True
+                return False
```

Comment:
```suggestion
                return isinstance(result, cls)
```

Maybe somethinig like this?